identity/did/identity.go

```diff
@@ -18,3 +29,128 @@ func NewDID(address common.Address) DID {
 func NewDIDFromString(address string) DID {
 	return DID(common.HexToAddress(address))
 }
+
+// Identity interface contains the methods to interact with the identity contract
+type Identity interface {
+	// AddKey adds a key to identity contract
+	AddKey(did *DID, key Key) (chan *ethereum.WatchTransaction, error)
+	// GetKey return a key from the identity contract
+	GetKey(did *DID, key [32]byte) (*KeyResponse, error)
+}
+
+type contract interface {
+
+	// calls
+	GetKey(opts *bind.CallOpts, _key [32]byte) (struct {
+		Key       [32]byte
+		Purposes  []*big.Int
+		RevokedAt *big.Int
+	}, error)
+
+	// transactions
+	AddKey(opts *bind.TransactOpts, _key [32]byte, _purpose *big.Int, _keyType *big.Int) (*types.Transaction, error)
+}
+
+type identity struct {
+	contract contract
+	config   id.Config
+	client   ethereum.Client
+}
+
+func (i identity) prepareTransaction(did DID) (contract, *bind.TransactOpts, error) {
+	opts, err := i.client.GetTxOpts(i.config.GetEthereumDefaultAccountName())
+	if err != nil {
+		log.Infof("Failed to get txOpts from Ethereum client: %v", err)
+		return nil, nil, err
+	}
+
+	contract, err := i.bindContract(did)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return contract, opts, nil
+
+}
+
+func (i identity) prepareCall(did DID) (contract, *bind.CallOpts, context.CancelFunc, error) {
+	opts, cancelFunc := i.client.GetGethCallOpts()
+
+	contract, err := i.bindContract(did)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	return contract, opts, cancelFunc, nil
+
+}
+
+func (i identity) bindContract(did DID) (contract, error) {
+	contract, err := NewIdentityContract(did.toAddress(), i.client.GetEthClient())
+	if err != nil {
+		return nil, errors.New("Could not bind identity contract: %v", err)
+	}
+
+	return contract, nil
+
+}
+
+// NewIdentity creates a instance of an identity
+func NewIdentity(config id.Config, client ethereum.Client) Identity {
+	return identity{config: config, client: client}
+}
+
+// TODO: will be replaced with statusTask
+func waitForTransaction(client ethereum.Client, txHash common.Hash, txStatus chan *ethereum.WatchTransaction) {
```

Comment:
waiting for #663 to get merged :)

Reply:
@manuelpolzhofer You can merge this without that PR using the older way of doing async once comments are addressed. I have a design for that that changes things for this and anchoring anyways. The design basically introduces the concept of a TX manager. Instead of tasks managing transactions.